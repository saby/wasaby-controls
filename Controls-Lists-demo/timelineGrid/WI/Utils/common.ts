import { TNavigationDirection } from 'Controls/interface';
import { Quantum } from 'Controls-Lists/timelineGrid';

export function isWeekend(date: Date): boolean {
    const day = date.getDay();
    const SATURDAY = 6;
    const SUNDAY = 0;
    return day === SATURDAY || day === SUNDAY;
}

export function shiftDate(date: Date, direction: TNavigationDirection, quantum: Quantum): void {
    const shiftSize = direction === 'backward' ? -1 : 1;
    switch (quantum) {
        case Quantum.Second:
        case Quantum.Minute:
        case Quantum.Hour:
            date.setHours(date.getHours() + shiftSize);
            break;
        case Quantum.Day:
            date.setDate(date.getDate() + shiftSize);
            break;
        case Quantum.Month:
            date.setMonth(date.getMonth() + shiftSize);
            break;
    }
}
