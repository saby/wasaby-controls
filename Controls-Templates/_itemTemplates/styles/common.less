.Controls-Templates-TileItem__wrapper {
   overflow: hidden;
   box-sizing: border-box;
   user-select: none;
}

.Controls-Templates-TileItem__wrapper-background-default {
   background-color: var(--background-color);
}

.Controls-Templates-TileItem__image_fill-item {
   align-items: center;
   justify-content: center;
   overflow: hidden;
   z-index: -1;
}

.Controls-Templates-TileItem__content {
   display: flex;
   position: relative;
   width: 100%;
   flex-grow: 1;
}

.Controls-Templates-TileItem__image_size_half {
   width: 50%;
}

.Controls-Templates-TileItem__image_size {
   /* stylelint-disable-next-line unit-no-unknown */
   @sizes: s, m, l, xl, xlt, 2xl, 3xl, 4xl, 5xl, 6xl, 7xl;
   .for(@sizes, {
      &_@{_item} {
         width: ~'var(--size_@{_item}_image)';
         height: ~'var(--size_@{_item}_image)';
      }
   });
}

.Controls-Templates-TileItem__image {
   max-width: 100%;
   max-height: 100%;
   width: auto;
   height: auto;
   background-repeat: no-repeat;
   box-sizing: border-box;
}

.Controls-Templates-TileItem__image_centered {
   position: absolute;
   top: 50%;
   left: 50%;
   transform: translate(-50%, -50%);
   background-position: center;
}

.Controls-Templates-TileItem__content_color-background {
   background-color: var(--dominant-color_zen);
}

.Controls-Templates-TileItem__image_bordered {
   border: var(--border-thickness) solid var(--unaccented_border-color);
}

.Controls-Templates-TileItem__content_as_column {
   display: flex;
   flex-direction: column;
   justify-content: inherit;
   height: 100%;
   width: 100%;
   box-sizing: border-box;
   min-width: 0;
   min-height: 0;
}

.Controls-Templates-TileItem__absolute-content {
   position: absolute;
   left: 0;
   right: 0;
   z-index: 1;
}

.Controls-Templates-TileItem__absolute-header {
   top: 0;
}

.Controls-Templates-TileItem__absolute-footer {
   bottom: 0;
}

.Controls-Templates-TileItem__content_justify_top {
   justify-content: flex-start;
}

.Controls-Templates-TileItem__content_justify_bottom {
   justify-content: flex-end;
}

.Controls-Templates-TileItem__image_proportion_1_1 {
   padding-top: 100%;
}

.Controls-Templates-TileItem__image_proportion_4_3 {
   padding-top: 100% * 0.75;
}

.Controls-Templates-TileItem__image_proportion_2_1 {
   padding-top: 100% * 0.5;
}

.Controls-Templates-TileItem__image_proportion_1_2 {
   padding-top: 100% * 2;
}

.Controls-Templates-TileItem__image_proportion_3_4 {
   padding-top: 100% * 1.33;
}

.Controls-Templates-TileItem__image_proportion_16_9 {
   padding-top: 100% * 0.56;
}

.Controls-Templates-TileItem__image_object-fit {
   @object-fits: fill, contain, cover, none, scale-down;
   .for(@object-fits, {
      &_@{_item} {
         object-fit: @_item;
         background-size: @_item;
      }
   });
}

.Controls-Templates-TileItem__image_object-fit_cover {
   height: 100%;
   width: 100%;
}

.Controls-Templates-TileItem__wrapper .controls-itemActionsV {
   height: var(--inline_height_l);
   min-width: var(--inline_height_l);
   padding: 0;
   display: flex;
   justify-content: center;
}

.Controls-Templates-TileItem__wrapper .controls-itemActionsV_editing_translucent,
.Controls-Templates-TileItem__wrapper .controls-itemActionsV_style_translucent {
   background-color: var(--translucent_background-color);
}

.Controls-Templates-TileItem__wrapper .controls-itemActionsV_editing_translucent_light,
.Controls-Templates-TileItem__wrapper .controls-itemActionsV_style_translucent_light {
   background-color: var(--translucent_background-color_light);
}

.Controls-Templates-TileItem__wrapper:hover
   .Controls-Templates-BaseItem__itemActionsV_shadow_onhover,
.Controls-Templates-BaseItem__itemActionsV_shadow_visible {
   box-shadow: var(--item_actions_shadow_size_rich_tile) var(--item_actions_shadow_color_rich_tile);
}
