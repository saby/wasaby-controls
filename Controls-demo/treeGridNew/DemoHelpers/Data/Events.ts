import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
