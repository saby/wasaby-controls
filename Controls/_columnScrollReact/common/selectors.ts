const prefix = 'js-controls-ColumnScrollReact';
type _TSeparator = '_' | '';

export const ENABLE_GRAPHIC_ACCELERATION_CSS_CLASS_NAME =
    'controls-ColumnScrollReact__acceleration';

export const DEFAULT_SELECTORS = {
    ROOT_TRANSFORMED_ELEMENT:
        `${prefix}_root-scroll-transform` as `${typeof prefix}_root-scroll-transform`,
    FIXED_ELEMENT: `${prefix}_fixed` as `${typeof prefix}_fixed`,
    SCROLLABLE_ELEMENT: `${prefix}_scrollable` as `${typeof prefix}_scrollable`,
    NOT_DRAG_SCROLLABLE: `${prefix}_notDragScrollable` as `${typeof prefix}_notDragScrollable`,
    FIXED_TO_LEFT_EDGE_ELEMENT: `${prefix}_offsetForFixed` as `${typeof prefix}_offsetForFixed`,
    HIDE_ALL_FIXED_ELEMENTS: `${prefix}_hideFixedElements` as `${typeof prefix}_hideFixedElements`,
};

export interface ISelectorsProps {
    FIXED_ELEMENT?: string;
    SCROLLABLE_ELEMENT?: string;
    NOT_DRAG_SCROLLABLE?: string;
    ROOT_TRANSFORMED_ELEMENT?: string;
}

export interface ISelectorsState extends Required<ISelectorsProps> {
    FIXED_ELEMENT_STATIC: string;
    HIDE_ALL_FIXED_ELEMENTS: `${typeof DEFAULT_SELECTORS.HIDE_ALL_FIXED_ELEMENTS}${_TSeparator}${string}`;
    FIXED_TO_LEFT_EDGE_ELEMENT: `${typeof DEFAULT_SELECTORS.FIXED_TO_LEFT_EDGE_ELEMENT}${_TSeparator}${string}`;
}

export function getSelectorsState(
    propsSelectors: ISelectorsProps = {},
    guid: string = ''
): ISelectorsState {
    const requiredSelectors = _getPropsSelectors(propsSelectors);
    const separator: _TSeparator = guid ? '_' : '';

    return {
        FIXED_ELEMENT: `${requiredSelectors.FIXED_ELEMENT}${separator}${guid}`,
        SCROLLABLE_ELEMENT: `${requiredSelectors.SCROLLABLE_ELEMENT}${separator}${guid}`,
        NOT_DRAG_SCROLLABLE: `${requiredSelectors.NOT_DRAG_SCROLLABLE}${separator}${guid}`,
        ROOT_TRANSFORMED_ELEMENT: `${requiredSelectors.ROOT_TRANSFORMED_ELEMENT}${separator}${guid}`,

        FIXED_ELEMENT_STATIC: `${guid ? requiredSelectors.FIXED_ELEMENT : ''}`,
        HIDE_ALL_FIXED_ELEMENTS: `${DEFAULT_SELECTORS.HIDE_ALL_FIXED_ELEMENTS}${separator}${guid}`,
        FIXED_TO_LEFT_EDGE_ELEMENT: `${DEFAULT_SELECTORS.FIXED_TO_LEFT_EDGE_ELEMENT}${separator}${guid}`,
    };
}

function _getPropsSelectors(propsSelectors: ISelectorsProps): Required<ISelectorsProps> {
    return {
        FIXED_ELEMENT: propsSelectors.FIXED_ELEMENT || DEFAULT_SELECTORS.FIXED_ELEMENT,
        SCROLLABLE_ELEMENT:
            propsSelectors.SCROLLABLE_ELEMENT || DEFAULT_SELECTORS.SCROLLABLE_ELEMENT,
        NOT_DRAG_SCROLLABLE:
            propsSelectors.NOT_DRAG_SCROLLABLE || DEFAULT_SELECTORS.NOT_DRAG_SCROLLABLE,
        ROOT_TRANSFORMED_ELEMENT:
            propsSelectors.ROOT_TRANSFORMED_ELEMENT || DEFAULT_SELECTORS.ROOT_TRANSFORMED_ELEMENT,
    };
}
