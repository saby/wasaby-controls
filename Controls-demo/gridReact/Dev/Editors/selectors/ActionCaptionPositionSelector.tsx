import * as React from 'react';
import { TActionCaptionPosition } from 'Controls/itemActions';
import { correctValue } from 'Controls-demo/gridReact/Dev/Editors/Utils';

type TValue = TActionCaptionPosition | undefined;
interface IProps {
    value: TValue;
    onChange: (value: TValue) => void;
}

export function ActionCaptionPositionSelector(
    props: IProps
): React.ReactElement {
    return (
        <select
            value={props.value}
            onChange={(event) => {
                return props.onChange(correctValue(event.target.value));
            }}
        >
            <option value={undefined}>default</option>
            <option value={'right'}>right</option>
            <option value={'bottom'}>bottom</option>
            <option value={'none'}>none</option>
        </select>
    );
}
