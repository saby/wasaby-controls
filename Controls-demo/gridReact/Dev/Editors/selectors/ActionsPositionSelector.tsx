import * as React from 'react';
import { TItemActionsPosition } from 'Controls/itemActions';
import { correctValue } from 'Controls-demo/gridReact/Dev/Editors/Utils';

type TValue = TItemActionsPosition | undefined;
interface IProps {
    value: TValue;
    onChange: (value: TValue) => void;
}

export function ActionsPositionSelector(props: IProps): React.ReactElement {
    return (
        <select
            value={props.value}
            onChange={(event) => {
                return props.onChange(correctValue<TValue>(event.target.value));
            }}
        >
            <option value={undefined}>default</option>
            <option value={'inside'}>inside</option>
            <option value={'outside'}>outside</option>
            <option value={'custom'}>custom</option>
        </select>
    );
}
