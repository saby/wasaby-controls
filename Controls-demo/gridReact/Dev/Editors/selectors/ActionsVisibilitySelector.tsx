import * as React from 'react';
import { TItemActionsVisibility } from 'Controls/itemActions';
import { correctValue } from 'Controls-demo/gridReact/Dev/Editors/Utils';

type TValue = TItemActionsVisibility | undefined;
interface IProps {
    value: TValue;
    onChange: (value: TValue) => void;
}

export function ActionsVisibilitySelector(props: IProps): React.ReactElement {
    return (
        <select
            value={props.value}
            onChange={(event) => {
                return props.onChange(correctValue<TValue>(event.target.value));
            }}
        >
            <option value={undefined}>default</option>
            <option value={'hidden'}>hidden</option>
            <option value={'visible'}>visible</option>
            <option value={'onhover'}>onhover</option>
            <option value={'delayed'}>delayed</option>
        </select>
    );
}
