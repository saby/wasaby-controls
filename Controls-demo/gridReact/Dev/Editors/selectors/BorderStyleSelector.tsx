import * as React from 'react';

import { TBorderStyle } from 'Controls/display';
import { correctValue } from 'Controls-demo/gridReact/Dev/Editors/Utils';

type TValue = TBorderStyle | undefined;
interface IProps {
    value: TValue;
    onChange: (value: TValue) => void;
}

export function BorderStyleSelector(props: IProps): React.ReactElement {
    return (
        <select
            value={props.value}
            onChange={(event) => {
                return props.onChange(correctValue<TValue>(event.target.value));
            }}
        >
            <option value={undefined}>default</option>
            <option value={'danger'}>danger</option>
        </select>
    );
}
