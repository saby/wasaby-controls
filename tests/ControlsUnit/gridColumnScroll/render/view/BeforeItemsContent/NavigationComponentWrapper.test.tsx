/**
 * @jest-environment jsdom
 */
jest.mock('Controls/_columnScrollReact/NavigationComponent', () => {
    return {
        __esModule: true,
        default: (props) => {
            return (
                <div data-qa="mocked_ColumnScrollReact__NavigationComponent">
                    <div data-qa="mocked_ColumnScrollReact__NavigationComponent">
                        {JSON.stringify(props, null, 4)}
                    </div>
                </div>
            );
        },
    };
});
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';

import {
    InnerNavigationComponentWrapper,
    IInnerNavigationComponentWrapperProps,
} from 'Controls/_gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper';
import { DEFAULT_SELECTORS, getSelectorsState } from 'Controls/_columnScrollReact/common/selectors';

describe('ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    describe('Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn.', () => {
        const staticProps: IInnerNavigationComponentWrapperProps = {
            part: undefined,
            mode: undefined,
            isMobileScrolling: undefined,
            hasGridStickyHeader: undefined,
            startFixedWidth: 100,
            endFixedWidth: 0,
            contentWidth: 1000,
            viewPortWidth: 500,
            SELECTORS: getSelectorsState(DEFAULT_SELECTORS),
        };
        const parts: IInnerNavigationComponentWrapperProps['part'][] = ['fixed', 'scrollable'];
        const modes: IInnerNavigationComponentWrapperProps['mode'][] = ['scrollbar', 'arrows'];
        const hasGridStickyHeaderVariants: IInnerNavigationComponentWrapperProps['hasGridStickyHeader'][] =
            [true, false];
        const isMobileScrollingVariants: IInnerNavigationComponentWrapperProps['isMobileScrolling'][] =
            [true, false];

        parts.forEach((part) => {
            modes.forEach((mode) => {
                hasGridStickyHeaderVariants.forEach((hasGridStickyHeader) => {
                    isMobileScrollingVariants.forEach((isMobileScrolling) => {
                        it(`part=${part}, mode=${mode}, hasGridStickyHeader=${hasGridStickyHeader}, isMobileScrolling=${isMobileScrolling}`, () => {
                            const { asFragment } = render(
                                <InnerNavigationComponentWrapper
                                    {...staticProps}
                                    part={part}
                                    mode={mode}
                                    hasGridStickyHeader={hasGridStickyHeader}
                                    isMobileScrolling={isMobileScrolling}
                                />,
                                {
                                    container,
                                }
                            );

                            expect(asFragment()).toMatchSnapshot();
                        });
                    });
                });
            });
        });
    });
});
