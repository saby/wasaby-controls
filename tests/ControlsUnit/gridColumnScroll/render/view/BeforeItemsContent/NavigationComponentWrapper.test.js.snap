// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=arrows, hasGridStickyHeader=false, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=arrows, hasGridStickyHeader=false, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=arrows, hasGridStickyHeader=true, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=arrows, hasGridStickyHeader=true, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=scrollbar, hasGridStickyHeader=false, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=scrollbar, hasGridStickyHeader=false, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=scrollbar, hasGridStickyHeader=true, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=fixed, mode=scrollbar, hasGridStickyHeader=true, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_fixed controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=arrows, hasGridStickyHeader=false, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=arrows, hasGridStickyHeader=false, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px; display: none;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=arrows, hasGridStickyHeader=true, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=arrows, hasGridStickyHeader=true, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_arrows"
    style="width: 400px; display: none;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "arrows",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=scrollbar, hasGridStickyHeader=false, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=scrollbar, hasGridStickyHeader=false, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px; display: none;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "bottom"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=scrollbar, hasGridStickyHeader=true, isMobileScrolling=false 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/NavigationComponentWrapper Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. part=scrollable, mode=scrollbar, hasGridStickyHeader=true, isMobileScrolling=true 1`] = `
<DocumentFragment>
  <div
    class="js-controls-ColumnScrollReact_fixed js-controls-GridColumnScroll_fixed controls-GridReact__columnScroll__headerNavigation controls-GridReact__columnScroll__headerNavigation-part_scrollable controls-GridReact__columnScroll__headerNavigation-mode_scrollbar"
    style="width: 400px; display: none;"
  >
    <div
      data-qa="mocked_ColumnScrollReact__NavigationComponent"
    >
      <div
        data-qa="mocked_ColumnScrollReact__NavigationComponent"
      >
        {
    "mode": "scrollbar",
    "scrollBarValign": "center"
}
      </div>
    </div>
  </div>
</DocumentFragment>
`;
